src/main/java/org/jabref/gui/strings/StringDialogController.java

```diff
@@ -0,0 +1,122 @@
+
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableColumn.CellEditEvent;
+import javafx.scene.control.TableView;
+import javafx.scene.control.Tooltip;
+import javafx.scene.control.cell.TextFieldTableCell;
+import javafx.util.converter.DefaultStringConverter;
+
+import org.jabref.gui.AbstractController;
+import org.jabref.gui.IconTheme.JabRefIcons;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.help.HelpAction;
+import org.jabref.gui.util.IconValidationDecorator;
+import org.jabref.logic.help.HelpFile;
+import org.jabref.logic.l10n.Localization;
+
+import de.saxsys.mvvmfx.utils.validation.visualization.ControlsFxVisualizer;
+import org.controlsfx.tools.ValueExtractor;
+
+public class StringDialogController extends AbstractController<StringDialogViewModel> {
+
+    @FXML private Button btnNewString;
+    @FXML private Button btnRemove;
+    @FXML private Button btnHelp;
+    @FXML private Button btnCancel;
+    @FXML private Button btnSave;
+    @FXML private TableView<StringViewModel> tblStrings;
+    @FXML private TableColumn<StringViewModel, String> colLabel;
+    @FXML private TableColumn<StringViewModel, String> colContent;
+    @Inject private StateManager stateManager;
+
+    @FXML
+    private void initialize() {
+
+        viewModel = new StringDialogViewModel(stateManager);
+
+        btnHelp.setGraphic(JabRefIcons.HELP.getGraphicNode());
+        btnHelp.setTooltip(new Tooltip(Localization.lang("Open Help page")));
+
+        btnNewString.setGraphic(JabRefIcons.ADD.getGraphicNode());
+        btnNewString.setTooltip(new Tooltip(Localization.lang("New string")));
+
+        btnRemove.setGraphic(JabRefIcons.REMOVE.getGraphicNode());
+        btnRemove.setTooltip(new Tooltip(Localization.lang("Remove string")));
+        ControlsFxVisualizer visualizer = new ControlsFxVisualizer();
+        visualizer.setDecoration(new IconValidationDecorator());
+
+        //Register TextfieldTableCell Control for validation
+        ValueExtractor.addObservableValueExtractor(c -> c instanceof TextFieldTableCell, c -> ((TextFieldTableCell<?, String>) c).textProperty());
+
+        colLabel.setCellFactory(column -> {
+
+            TextFieldTableCell<StringViewModel, String> cell = new TextFieldTableCell<>(new DefaultStringConverter());
+            column.setCellValueFactory(cellData -> {
+
+                visualizer.initVisualization(cellData.getValue().labelValidation(), cell);
+                return cellData.getValue().getLabel();
+            });
+            return cell;
+
+        });
+
+        colContent.setCellFactory(column -> {
+
+            TextFieldTableCell<StringViewModel, String> cell = new TextFieldTableCell<>(new DefaultStringConverter());
+            column.setCellValueFactory(cellData -> {
+
+                visualizer.initVisualization(cellData.getValue().contentValidation(), cell);
```

Comment:
Two ideas:
1. Is this code called when you enter data in the text field? If you hit enter and leave the edit mode?
2. Is the visualizer correctly installed (i.e. debug the `initVisualization` code). Maybe it is expecting the text field and not a cell...

Reply:
Yes, I debugged it and it's called everytime. I also debugged through the whole code from controlsfx and have no idea what might be wrong.
I then thought it might be that the css class is missing, so I copied the stuff from the entry editor css, but still does not work.

The Textfield TableCell was working previously with the default settings after I registered the ValueExtractor for controlsfx. But this is no longer used and hasn't any effect.